### ISSUE TITLE
full path of compilation directory inside stripped executable

### ISSUE THREAD

**Issue by pirate486743186** [TerryCavanagh/VVVVVV#889]
title...

This happens at least in linux. It shouldn't happen.

```
grep -Fr "full path compilation dir" VVVVVV
grep: VVVVVV: binary file matches
```

compiled with `cmake -DOFFICIAL_BUILD=ON` then stripped.

**Comment by InfoTeddy** [comment]
This seems to be because of all the `SDL_assert`s getting compiled in to the binary.

**Comment by InfoTeddy** [comment]
The `SDL_assert`s go away if you compile with any level of optimization (e.g. `-O1` and above). But it looks like this still happens with PhysFS and FAudio. FAudio seems like it's using `SDL_assert` (because we don't have `FAUDIO_LOG_ASSERTIONS` defined) but then in that case it shouldn't be compiled in when compiling with `-O1` and such (@flibitijibibo any ideas?); PhysFS looks like it uses straight `assert`s so I'm not sure how to get rid of those. Also worth noting is that the official shipped binaries on Steam, GOG etc. don't seem to have this problem either, so I'm not sure what's different.

I guess now is as good a time to ask... what exactly is the problem with having the compilation directory inside the binary? It seems harmless to me.

**Comment by InfoTeddy** [comment]
> Also people can care about reproducible builds.

Ooh, yeah, I kinda want reproducible builds too, I just haven't gotten around to doing the work for them yet.

> At line 311 add this, it works for me, i don't know if it's general enough.
> 
> ```
> get_filename_component(REPO_DIR ../ ABSOLUTE)
> target_compile_options(physfs-static PRIVATE "-ffile-prefix-map=${REPO_DIR}=.")
> ```

So you mean a patch that looks like this?

```diff
diff --git a/desktop_version/CMakeLists.txt b/desktop_version/CMakeLists.txt
index 55a53186..b255ba28 100644
--- a/desktop_version/CMakeLists.txt
+++ b/desktop_version/CMakeLists.txt
@@ -309,6 +309,8 @@ if(BUNDLE_DEPENDENCIES)

     target_link_libraries(VVVVVV physfs-static tinyxml2-static lodepng-static faudio-static)
 else()
+    get_filename_component(REPO_DIR ../ ABSOLUTE)
+    target_compile_options(physfs-static PRIVATE "-ffile-prefix-map=${REPO_DIR}=.")
     find_package(utf8cpp CONFIG)

     target_link_libraries(VVVVVV physfs tinyxml2 utf8cpp lodepng-static FAudio)
```

It doesn't seem to work - `ninja` says that there's no work to do. It still produces a binary with the paths even if I nuke the `CMakeFiles` and `CMakeCache.txt` and do a fresh build.

**Comment by InfoTeddy** [comment]
> it's a unix only fix.

![2022-08-20_11-41-00](https://user-images.githubusercontent.com/59748578/185761716-ffce6784-4741-433e-aabf-6d623727385c.png)

**Comment by InfoTeddy** [comment]
So I think I figured out what the problem is... FAudio!

Putting aside the fact that PhysFS asserts don't seem to go away even in Release mode and I had to add `target_compile_options(physfs-static PRIVATE -DNDEBUG)`, the remaining paths come from FAudio macros, which look like they use `__FILE__` directly with no way to turn them off in release builds without modifying the source... @flibitijibibo, is this something you'd be interested in rectifying?

At any rate it seems like `-ffile-prefix-map` works just fine for getting rid of the paths and turning all potentially sensitive information into periods, it just needs to be applied to FAudio too. It also works on Clang. It doesn't seem like there's an equivalent for MSVC, [judging by the page on reproducible-builds.org](https://reproducible-builds.org/docs/build-path/), but who cares about Windows anyways.
